src/main/java/com/minecolonies/coremod/colony/managers/BuildingManager.java

```diff
@@ -257,6 +257,34 @@ public IBuilding getBuilding(final BlockPos buildingId)
         return null;
     }
 
+    @Nullable
+    @Override
+    public IWareHouse getClosestWarehouseInColony(final BlockPos pos)
+    {
+        final List<IWareHouse> wareHouses = new ArrayList<>();
+        for (final IBuilding building : colony.getBuildingManager().getBuildings().values())
+        {
+            if (building instanceof BuildingWareHouse && building.getTileEntity() != null)
+            {
+                wareHouses.add((IWareHouse) building);
+            }
+        }
+
+        wareHouses.removeIf(Objects::isNull);
+        wareHouses.sort((w1, w2) ->
+        {
+            final double dist1 = w1.getPosition().distanceSq(pos);
+            final double dist2 = w2.getPosition().distanceSq(pos);
+            return Double.compare(dist1, dist2);
+        });
```

Comment:
if you only need the closest just keep the closest saved, no need to re-sort the entire list